src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/AbstractRazorEditorTest.cs

```diff
@@ -96,19 +100,20 @@ private async Task<string> CreateAndOpenBlazorProjectAsync(CancellationToken can
 
         var solutionPath = CreateTemporaryPath();
 
-        var resourceName = "Microsoft.VisualStudio.Razor.IntegrationTests.TestFiles.BlazorProject.zip";
-        using var zipStream = typeof(AbstractRazorEditorTest).Assembly.GetManifestResourceStream(resourceName);
+        using var zipStream = typeof(AbstractRazorEditorTest).Assembly.GetManifestResourceStream(ProjectZipFile);
         using var zip = new ZipArchive(zipStream);
         zip.ExtractToDirectory(solutionPath);
 
         var slnFile = Directory.EnumerateFiles(solutionPath, "*.sln").Single();
-        var projectFile = Directory.EnumerateFiles(solutionPath, "*.csproj", SearchOption.AllDirectories).Single();
 
-        PrepareProjectForFirstOpen(projectFile);
+        foreach (var projectFile in Directory.EnumerateFiles(solutionPath, "*.csproj", SearchOption.AllDirectories))
+        {
+            PrepareProjectForFirstOpen(projectFile);
+        }
 
         await TestServices.SolutionExplorer.OpenSolutionAsync(slnFile, cancellationToken);
 
-        return projectFile;
+        return Directory.EnumerateFiles(solutionPath, $"{RazorProjectConstants.BlazorProjectName}.csproj", SearchOption.AllDirectories).Single();
```

Comment:
I know this is how it was before, but why are we enumerating for just one? Or is this to assert that there is ONLY one?

Reply:
This method did two things: Find the one "main" project we're using and return its path, and update the project file to the right TFM.

This enumeration is just the first bit, but we now also need to enumerate all of the projects to update all of the TFMs. I could have checked for this file in the other enumeration and saved it to a local but I am especially lazy in integration tests.